src/cobovault.js

```diff
@@ -0,0 +1,313 @@
+import {parseSignaturesFromPSBT, unsignedMultisigPSBT} from "unchained-bitcoin";
+
+export const COBOVAULT = 'cobovault';
+
+import {
+    IndirectKeystoreInteraction,
+    PENDING,
+    ACTIVE,
+    INFO,
+} from "./interaction";
+
+import {encodeUR, decodeUR, extractSingleWorkload} from '@cvbb/bc-ur'
+
+function onlyUniq(value, index, self) {
+    return self.indexOf(value) === index;
+}
+
+/**
+ * @param {array<string>} qrs - the PSBT
+ * @returns {Object} parsed result
+ * {
+ *     success: boolean,
+ *     result: string | object
+ *     type: 'text' | 'ur' | 'json'
+ *     current: number,
+ *     total: number,
+ *     workloads: array<string>
+ * }
+ */
+function smartParseQR(qrs) {
+    const enhancedQRS = qrs.filter(onlyUniq);
+    const first = qrs[0];
```

Comment:
Would it not be better to use `enhancedQRs[0]` here instead of `qrs[0]` ?

Reply:
👍